apis/clusterresources/v1beta1/opensearchegressrules_webhook.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"github.com/instaclustr/operator/pkg/models"
+)
+
+// log is for logging in this package.
+var opensearchegressruleslog = logf.Log.WithName("opensearchegressrules-resource")
+
+var openSearchBindingIDPattern, _ = regexp.Compile(`[\w-]+`)
+
+func (r *OpenSearchEgressRules) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-clusterresources-instaclustr-com-v1beta1-opensearchegressrules,mutating=false,failurePolicy=fail,sideEffects=None,groups=clusterresources.instaclustr.com,resources=opensearchegressrules,verbs=create;update,versions=v1beta1,name=vopensearchegressrules.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &OpenSearchEgressRules{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *OpenSearchEgressRules) ValidateCreate() error {
+	opensearchegressruleslog.Info("validate create", "name", r.Name)
+
+	if r.Spec.ClusterID == "" || r.Spec.OpenSearchBindingID == "" || r.Spec.Source == "" {
+		return fmt.Errorf("clusterId, openSearchBindingId, source must be filled")
+	}
```

Comment:
You also need to add enum validation for `source`: `"ALERTING"`, `"NOTIFICATIONS"`
And for `type`: `"SLACK"`, `"WEBHOOK"`, `"CUSTOM_WEBHOOK"`, `"CHIME"`

Reply:
I think it`s unnecessary, all such validations implemented on Instaclustr API, and in case they add new types, logic wouldn't work. About "ALERTING" it`s already deprecated, I`d rather avoid such validations if it's possible

Reply:
Yea, but in such way it will be created in k8s but wasn't created in Instaclustr.